lib/babe/babe.go

```diff
@@ -336,158 +303,105 @@ func (b *Service) initiate() {
 		return
 	}
 
-	err := b.invokeBlockAuthoring()
-	if err != nil {
-		logger.Criticalf("block authoring error: %s", err)
+	// we should consider better error handling for this - we should
+	// retry to run the engine at some point (maybe the next epoch) if
+	// there's an error.
+	if err := b.runEngine(); err != nil {
+		logger.Criticalf("failed to run block production engine: %s", err)
 	}
 }
 
-func (b *Service) invokeBlockAuthoring() error {
-	epoch, err := b.epochState.GetCurrentEpoch()
+func (b *Service) initiateAndGetEpochHandler(epoch uint64) (*epochHandler, error) {
+	epochData, err := b.initiateEpoch(epoch)
 	if err != nil {
-		logger.Errorf("failed to get current epoch: %s", err)
-		return err
+		return nil, fmt.Errorf("failed to initiate epoch %d: %s", epoch, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to initiate epoch %d: %w", epoch, err)
```

Reply:
done (also fixed all the other error returns)